sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_shared/models.py

```diff
@@ -2,14 +2,20 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
 # ------------------------------------
-from enum import Enum
+from enum import Enum, EnumMeta
 from typing import Mapping, Optional, Union, Any
-
+import warnings
 from typing_extensions import TypedDict, Protocol
 from azure.core import CaseInsensitiveEnumMeta
 
+class DeprecatedEnumMeta(CaseInsensitiveEnumMeta):
+
+    def __getattribute__(cls, item):
+        if item == "MICROSOFT_BOT":
```

Comment:
This still needs to actually return something.
Once the bot identifier finally goes GA, what name will we be using in the enum?

Reply:
we will add `MICROSOFT_TEAMS_APP = "microsoft_teams_app"` and in GA version we can return MICROSOFT_TEAMS_APP for MICROSOFT_BOT item request

Reply:
We still need to return something in the meantime. This will currently print a warning then immediately raise an AttributeError

Reply:
It's still returning CommunicationIdentifierKind.MICROSOFT_BOT after printing the deprecation error. I added even assertion for it([test_deprecated_bot_kind](https://github.com/Azure/azure-sdk-for-python/blob/00050543006469e7a3c865df30ea22421145865e/sdk/communication/azure-communication-identity/tests/test_deprecations.py#L20)), we should add return statement when we will introduce MICROSOFT_TEAMS_APP and return it.

Reply:
Ah you're right - sorry, I thought you had removed MICROSOFT_BOT from the public enum altogether.